### PR TITLE
Remove basic auth for GKE

### PR DESCRIPTION
**Pull request by thedadams**
The basic was used to generate an access token. However, the access
token was also being generated when getting a service client. Instead of
using the basic auth, we now use the access token.

Issues:
https://github.com/rancher/rancher/issues/29587
https://github.com/rancher/rancher/issues/28673

### CODE REVIEW

**Review by rmweir — APPROVED** [review]
LGTM, looks clean great job 👍

**Review by prachidamle — DISMISSED** [review]
lgtm

File: pkg/kontainer-engine/drivers/gke/gke_driver.go [hunk 1]
```diff
@@ -786,7 +787,12 @@ func (d *Driver) PostCheck(ctx context.Context, info *types.ClusterInfo) (*types
 		return nil, err
 	}
 
-	svc, err := d.getServiceClient(ctx, state)
+	ts, err := getTokenSource(ctx, state)
+	if err != nil {
+		return nil, err
+	}
+
+	svc, err := getServiceClientWithTokenSource(ctx, ts)
```
[the remark this answers is not in the document]
Comment (thedadams) [reply pkg/kontainer-engine/drivers/gke/gke_driver.go:795]: You are correct: `getServiceClientWithTokenSource` was created for this exact purpose.

File: pkg/kontainer-engine/drivers/gke/gke_driver.go [hunk 2]
```diff
@@ -841,7 +847,7 @@ func (d *Driver) Remove(ctx context.Context, info *types.ClusterInfo) error {
 	return nil
 }
 
-func (d *Driver) getServiceClient(ctx context.Context, state state) (*raw.Service, error) {
+func getTokenSource(ctx context.Context, state state) (oauth2.TokenSource, error) {
```
[the remark this answers is not in the document]
Comment (thedadams) [reply pkg/kontainer-engine/drivers/gke/gke_driver.go:850]: I removed this because the driver wasn't being used at all. Was just simplifying.

**Review by cbron — APPROVED** [review]
lgtm, we can merge into master and backport (but not merge) to 2.5, I'll ask about 2.4.

**Pull request merged**